creator-node/src/components/replicaSet/replicaSetController.js

```diff
@@ -29,8 +39,65 @@ const respondToURSMRequestForProposalController = async (req) => {
   }
 }
 
+/**
+ * Given walletPublicKeys array and target creatorNodeEndpoint, will request export
+ * of all user data, update DB state accordingly, fetch all files and make them available.
+ *
+ * This route is only run on secondaries, to export and sync data from a user's primary.
+ *
+ * @notice Returns success regardless of sync outcome -> primary node will re-request sync if needed
+ */
+const syncRouteController = async (req, res) => {
+  req.logger.info(`SIDTEST ENTER SYNCROUTECONTROLLER`)
+  const serviceRegistry = req.app.get('serviceRegistry')
+  const nodeConfig = serviceRegistry.nodeConfig
+
+  const walletPublicKeys = req.body.wallet // array
+  const creatorNodeEndpoint = req.body.creator_node_endpoint // string
+  const immediate = (req.body.immediate === true || req.body.immediate === 'true') // boolean
+  const blockNumber = req.body.blockNumber // integer
+
+  // Disable multi wallet syncs for now since in below redis logic is broken for multi wallet case
+  if (walletPublicKeys.length === 0) {
+    return errorResponseBadRequest(`Must provide one wallet param`)
+  } else if (walletPublicKeys.length > 1) {
+    return errorResponseBadRequest(`Multi wallet syncs are temporarily disabled`)
+  }
+
+  // If sync_type body param provided, log it (param is currently only used for logging)
+  const syncType = req.body.sync_type
+  if (syncType) {
+    req.logger.info(`SnapbackSM sync of type: ${syncType} initiated for ${walletPublicKeys} from ${creatorNodeEndpoint}`)
+  }
+
+  // Enqueue sync immediately or with debounce, per request param
+  if (immediate) {
+    await enqueueSync({ serviceRegistry, walletPublicKeys, creatorNodeEndpoint })
+  } else {
+    const debounceTime = nodeConfig.get('debounceTime')
+
+    for (let wallet of walletPublicKeys) {
+      if (wallet in syncQueue) {
+        clearTimeout(syncQueue[wallet])
+        req.logger.info('SyncRouteController - clear timeout for', wallet, 'time', Date.now())
+      }
+      syncQueue[wallet] = setTimeout(
+        async function () {
+          // return _nodesync(serviceRegistry, req.logger, [wallet], creatorNodeEndpoint, req.body.blockNumber, req.logContext)
+          await enqueueSync({ serviceRegistry, walletPublicKeys: [wallet], creatorNodeEndpoint, blockNumber })
+        },
```

Comment:
potentially should clear timeout here, maybe just leave a comment

Reply:
added timeout, good catch